File: CinemaBookingWeb/src/app/models/BookShow/bookshow.ts
import { Genre } from "../genre/genre"
import { Seat } from "../seat/seat"
import { Movie } from "../movie/movie"


export class BookShow
{
    hallName?: string;
    cinemaName?: string;
    movie?: Movie;
    seats?: Seat[];
    showId?: number;
    showDateTime?: string;
    selectedSeats?: Seat[];
    availableSeats?: Seat[];
    price?: number;
}

File: H3CinemaBooking/H3CinemaBooking.Repository/Models/BookShow.cs
using H3CinemaBooking.Repository.Models.DTO;
using Microsoft.CodeAnalysis;
using System.ComponentModel.DataAnnotations;
using System.Globalization;
using System.Text.Json.Serialization;

namespace H3CinemaBooking.Repository.Models
{
    public class BookShow
    {
        public string? HallName { get; set; }
        public string? CinemaName { get; set; }

        public Movie? Movie { get; set; }

        public DateTime ShowDateTime { get; set; }

        public List<SeatDTO>? Seats { get; set; } = new List<SeatDTO>();
        public double? Price { get; set; }
        public int? showId { get; set; }

    }


}
